sdk/communication/communication-jobrouter/review/communication-jobrouter.api.md

```diff
@@ -0,0 +1,693 @@
+// @public
+export interface QueueSelector {
+    key: string;
+    labelOperator: LabelOperator;
+    value?: Record<string, unknown>;
+}
+
+// @public
+export interface QueueSelectorAttachment {
+    kind: "conditional" | "pass-through" | "rule-engine" | "static" | "weighted-allocation-queue-selector";
+}
+
+// Warning: (ae-forgotten-export) The symbol "ConditionalQueueSelectorAttachment" needs to be exported by the entry point index.d.ts
+// Warning: (ae-forgotten-export) The symbol "PassThroughQueueSelectorAttachment" needs to be exported by the entry point index.d.ts
+// Warning: (ae-forgotten-export) The symbol "RuleEngineQueueSelectorAttachment" needs to be exported by the entry point index.d.ts
+// Warning: (ae-forgotten-export) The symbol "StaticQueueSelectorAttachment" needs to be exported by the entry point index.d.ts
+// Warning: (ae-forgotten-export) The symbol "WeightedAllocationQueueSelectorAttachment" needs to be exported by the entry point index.d.ts
+//
+// @public (undocumented)
+export type QueueSelectorAttachmentUnion = QueueSelectorAttachment | ConditionalQueueSelectorAttachment | PassThroughQueueSelectorAttachment | RuleEngineQueueSelectorAttachment | StaticQueueSelectorAttachment | WeightedAllocationQueueSelectorAttachment;
+
+// @public
+export interface QueueWeightedAllocation {
+    labelSelectors: QueueSelector[];
+    weight: number;
+}
+
+// @public
+export interface ReclassifyJobOptions extends JobRouterReclassifyJobActionOptionalParams {
+    reclassifyJobRequest?: Record<string, unknown>;
+}
+
+// @public
+export interface RoundRobinMode extends DistributionMode {
+    kind: "round-robin";
+}
+
+// @public
+export class RouterAdministrationClient {
+    constructor(connectionString: string, routerAdministrationClientOptions?: RouterAdministrationClientOptions);
+    constructor(endpoint: string, credential: KeyCredential, routerAdministrationClientOptions?: RouterAdministrationClientOptions);
+    constructor(endpoint: string, credential: TokenCredential, routerAdministrationClientOptions?: RouterAdministrationClientOptions);
+    constructor(endpoint: string, credential: CommunicationTokenCredential, routerAdministrationClientOptions?: RouterAdministrationClientOptions);
+    createClassificationPolicy(classificationPolicyId: string, options?: CreateClassificationPolicyOptions): Promise<ClassificationPolicy>;
+    createDistributionPolicy(distributionPolicyId: string, options?: CreateDistributionPolicyOptions): Promise<DistributionPolicy>;
+    createExceptionPolicy(exceptionPolicyId: string, options?: CreateExceptionPolicyOptions): Promise<ExceptionPolicy>;
+    createQueue(queueId: string, options?: CreateQueueOptions): Promise<JobQueue>;
+    deleteClassificationPolicy(classificationPolicyId: string): Promise<void>;
+    deleteDistributionPolicy(distributionPolicyId: string): Promise<void>;
+    deleteExceptionPolicy(exceptionPolicyId: string): Promise<void>;
+    deleteQueue(queueId: string): Promise<void>;
```

Comment:
It is from API review, that we should remove empty options, and we can always add them later as non-breaking changes.

Reply:
If there are no options specific for this operation we should at least send the regular request opetions. This should be

```ts
deleteQueue(queueId: string, options?: OperationOptions): Promise<void>;
```

Later on new can add specific options by extending OperationOptions.

We can't expose an operation without operation options, otherwise our customers will lack control over the configurable parts of each request, such as max retries.

Reply:
This applies for any method here with no options